File: apps/ai-app/app/router/openai.py
from app.core.response import ErrorResponse, SuccessResponse
from app.services.dependency import DepGreetingService
from fastapi import APIRouter

router = APIRouter(
    prefix="/openai",
    tags=["openai"],
)


@router.get(
    "/greetings",
    responses={
        200: {"model": SuccessResponse[dict]},
        400: {"model": ErrorResponse},
        502: {"model": ErrorResponse},
        503: {"model": ErrorResponse},
    },
)
async def greetings(service: DepGreetingService):
    greetings = await service.greetings()
    return greetings

File: apps/rust-ai/src/router/openai.rs
//! `/openai/greetings` route.
//!
//! Mirrors `app/router/openai.py`. Returns the `SuccessResponse` envelope
//! produced by `services::greeting::GreetingService::greetings()`.
//!
//! Span: `route.greetings` with attributes `http.method=GET` and
//! `http.route=/greetings`. This is REPLICABLE — must exact-match.

use axum::extract::State;
use axum::response::Response;
use axum::routing::get;
use tracing::Instrument;

use crate::AppState;
use crate::core::exception::AppError;

pub fn router() -> axum::Router<AppState> {
    axum::Router::new().route("/openai/greetings", get(greetings))
}

async fn greetings(State(state): State<AppState>) -> Result<Response, AppError> {
    let span = tracing::info_span!(
        "route.greetings",
        http.method = "GET",
        http.route = "/greetings"
    );
    state.greeting_service.greetings().instrument(span).await
}
